File: Algorithm/searching/python-code/LinearSearch.py
def linear_search(arr, target):
    for i in range(len(arr)):
        if arr[i] == target:
            return i  # found, return index
    return -1  # not found


if __name__ == "__main__":
    arr = [10, 20, 30, 40, 50]
    target = 30
    result = linear_search(arr, target)

    if result != -1:
        print(f"Element found at index {result}")
    else:
        print("Element not found")

File: Algorithm/searching/Linear_Search.cpp
#include <iostream>
#include <vector>
using namespace std;

int linearSearch(vector<int>& arr, int target) {
    for (int i = 0; i < arr.size(); i++) {
        if (arr[i] == target) return i; // found, return index
    }
    return -1; // not found
}

int main() {
    vector<int> arr = {10, 20, 30, 40, 50};
    int target = 30;
    int result = linearSearch(arr, target);

    if (result != -1)
        cout << "Element found at index " << result << endl;
    else
        cout << "Element not found" << endl;

    return 0;
}
